### PR TITLE
fix kwsplat check

### PR DESCRIPTION
**Pull request by aviatesk**
fix Screen Shot 2020-07-13 at 2 08 08

### CODE REVIEW

File: src/linting/checks.jl [hunk 1]
```diff
@@ -230,17 +229,24 @@ end
...
+        !(ref_minargs <= act_minargs <= act_maxargs <= ref_maxargs) && return false
     end
+
+    # check matching on keyword arguments
+    kwsplat && return true # splatted kw in method so accept any kw in call
+
+    # no splatted kw in method sig
+    length(act_kws) > length(ref_kws) && return false # call has more kws than method accepts
+    !all(kw in ref_kws for kw in act_kws) && return false # call supplies a kw that isn't defined in the method
+
```
Comment (aviatesk) [inline src/linting/checks.jl:249]: These don't change any logic, I ended up refactoring code a bit while trying to understand the logic. I will revert this if you want to keep the changes to be minimum

Comment (davidanthoff) [reply src/linting/checks.jl:249]: Yeah, I think in the current phase we should not do refactorings. I know, painful, I've had a couple of situations where I had a fix that changed unrelated things, and then I went back and boiled it down to only the fix...

Comment (aviatesk) [reply src/linting/checks.jl]: okay, this part is now separated into #208

File: src/linting/checks.jl [hunk 2]
```diff
@@ -149,21 +149,20 @@ function func_nargs(x::EXPR)
         if ispunctuation(arg)
             # skip
         elseif is_parameters(arg)
-            for j = 1:length(arg)
-                arg1 = arg[j]
-                if is_kwarg(arg1)
-                    push!(kws, Symbol(CSTParser.str_value(CSTParser.get_arg_name(arg1[1]))))
-                elseif is_binary_call(arg1) && kindof(arg1[2]) === CSTParser.Tokens.DDDOT
```
Comment (pfitzseb) [inline src/linting/checks.jl:156]: The "super minimal" fix would be to change this line to
```
elseif is_unary_call(arg1) && kindof(arg1[2]) === CSTParser.Tokens.DDDOT
```
right?

Comment (aviatesk) [reply src/linting/checks.jl:156]: yeah,  ...

**Comment by aviatesk** [comment]
I ended up doing a really minimum fix on #209 , and keep the original refactoring part in #208

**Pull request closed without merging**